lib/components/opaque_image_view.js

```diff
@@ -5,28 +5,55 @@ import React from 'react'
 import {
   processColor,
   requireNativeComponent,
-  ColorPropType,
   PixelRatio,
   StyleSheet
 } from 'react-native'
 
 import colors from '../../data/colors'
+import type { LayoutEvent } from '../system/events'
 
 const GeminiHost = 'd7hftxdivxxvm.cloudfront.net'
 const ImageQuality = 85
```

Comment:
OK, this one has some pretty big changes, but they're canonical for what larger classes _should_ look like, here's the overview:

``` javascript

// When dealing with larger prop/state declarations, move them
// into their own types 

type PropsType = { width: number }
type StateType = { height: ?number }

class Label extends React.Component {
  // they can then be used internally
  props: PropsType
  state: StateType

  // this then declares what the types of the defaults are too
  static defaultProps: {
    backgroundColor: string
  } 

  // There is now guarantees inside the constructor of types
  constructor(props: PropsType) {
    super(props)
    ...
  }

  // and in the render functions
  render() {
    return <View style={{width: this.props.width * 2}}>Label</div>;
  }
};

```